test/tests/functional/pbs_runjob_hook.py

```diff
@@ -113,6 +125,121 @@ def test_array_sub_new_res_in_hook(self):
             m = 'E;' + re.escape(sid) + ';.*Resource_List.site=site_value'
             self.server.accounting_match(m, regexp=True)
 
+    def test_array_sub_res_persist_in_hook_forcereque(self):
+        """
+        set custom resource in runjob hook. Submit a job array.
+        Check if custom resource set persists after force requeue due to
+        node fail requeue
+        """
+        # configure node fail requeue to lower value
+        self.server.manager(MGR_CMD_SET, SERVER,
+                            {'node_fail_requeue': 5})
+        # set three cpu for three subjobs
+        a = {'resources_available.ncpus': 3}
+        self.server.manager(MGR_CMD_SET, NODE, a, self.mom.shortname)
+        # create custom non-consumable string server resource
+        attr = {'type': 'string'}
+        r = 'foo_str'
+        self.server.manager(
+            MGR_CMD_CREATE, RSC, attr, id=r, logerr=False)
+        # create and import hook
+        hook_name = "runjob_hook"
+        attrs = {'event': "runjob"}
+        rv = self.server.create_import_hook(hook_name, attrs,
+                                            self.rerun_hook_script,
+                                            overwrite=True)
+        self.assertTrue(rv)
+        # create and submit job array
+        lower = 1
+        upper = 3
+        j1 = Job(TEST_USER)
+        j1.set_attributes({ATTR_J: '%d-%d' % (lower, upper)})
+        jid = self.server.submit(j1)
+        # check if running
+        self.server.expect(JOB, {ATTR_state: 'B'}, id=jid)
+        self.server.expect(JOB, {'job_state=R': 3}, count=True,
+                           id=jid, extend='t')
+        self.server.manager(MGR_CMD_SET, SERVER,
+                            {'scheduling': 'False'})
+        # kill mom
+        self.mom.stop('-KILL')
+        for i in range(lower, upper + 1):
+            sid = j1.create_subjob_id(jid, i)
+            m = "'runjob_hook' hook set job's "
+            "Resource_List.foo_str = foo_value"
+            self.server.tracejob_match(m, id=sid, n='ALL', tail=False)
+        # check subjob state change from R->Q
+        self.server.expect(JOB, {'job_state=Q': 3}, count=True,
+                           id=jid, extend='t')
+        # bring back mom
+        self.mom.start()
+        start_time = int(time.time())
+        self.mom.isUp()
+        # let subjobs get rerun from sched Q->R
+        self.server.manager(MGR_CMD_SET, SERVER,
+                            {'scheduling': 'True'})
+        self.server.expect(JOB, {'job_state=R': 3}, count=True,
+                           id=jid, extend='t')
+        # log match from hook log for custom res value
+        for i in range(lower, upper + 1):
+            sid = j1.create_subjob_id(jid, i)
+            m = "rerun_job_hook %s: Resource_List.foo_str=foo_value"
+            self.server.log_match(m % (sid), start_time)
+
+    def test_array_sub_res_persist_in_hook_qrerun(self):
```

Comment:
You have covered two rerun cases (node fail requeue and qrerun) there is a third one with preemption by requeue. It would be nice if you add a test for that scenario as well.

Reply:
I feel preemption by requeue scenario will be a redundant one as it wont be covering any new changes added through this PR, moreover the goal here is -not- to test different requeue scenarios.